### PR TITLE
Make travis fast again

### LINKED ISSUE #2307: Make Travis fast again !
Our Travis builds are hitting the limits again around 49 minutes.

I think we can start with a low hanging fruit by skipping some docker builds for things that we are not testing for.
For example we are building python and swift during recursive disDocker and again in running  [./gradlew :tests:testLean](https://github.com/apache/incubator-openwhisk/blob/master/tools/travis/build.sh#L33) task, this one excludes swift and python images  as part of test.dependencies.

### PR DESCRIPTION
**Pull request by csantanapr**
Closes #2307

### CODE REVIEW

File: tests/src/test/scala/system/basic/WskUnicodeJavaTests.scala [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import common.JsHelpers
+import common.TestHelpers
+import common.TestUtils
+import common.Wsk
+import common.WskProps
+import common.WskTestHelpers
+import spray.json._
+
+@RunWith(classOf[JUnitRunner])
+class WskUnicodeJavaTests
```
Comment (rabbah) [inline tests/src/test/scala/system/basic/WskUnicodeJavaTests.scala]: I'd make this an abstract class or a trait and have java extend it as well as the other tests.

File: tools/travis/build.sh [hunk 2]
```diff
@@ -12,6 +12,7 @@ tools/build/scanCode.py .
 cd $ROOTDIR/ansible
 
 ANSIBLE_CMD="ansible-playbook -i environments/local -e docker_image_prefix=testing"
+GRADLE_PROJS_SKIP="-x :core:pythonAction:distDocker  -x :core:python2Action:distDocker -x :core:swift3Action:distDocker"
```
Comment (rabbah) [inline tools/travis/build.sh]: Skip java too?

Comment (csantanapr) [inline tools/travis/build.sh]: Java currently being tested
I can add here but would need to update the testLean task to excluded all Java tests also.

File: tests/src/test/scala/system/basic/WskUnicodeJavaTests.scala (hunk 1 above)
Comment (csantanapr) [inline tests/src/test/scala/system/basic/WskUnicodeJavaTests.scala]: Can you help with this? And send me a PR to my branch.
 I modeled the thing after what we did at one point with with container tests for node012/6

Comment (csantanapr) [inline tests/src/test/scala/system/basic/WskUnicodeJavaTests.scala]: I think I figured it out the abstract

**Comment by csantanapr** [comment]
PG1 1583

**Pull request merged**